openpype/tools/attribute_defs/widgets.py

```diff
@@ -459,23 +459,23 @@ def current_value(self):
         idx = self._input_widget.currentIndex()
         return self._input_widget.itemData(idx)
 
+    def _multiselection_multivalue_prep(self, values):
+        final = None
+        are_same = True
+        for value in values:
+            value = set(value)
+            if final is None:
+                final = value
+            elif not are_same or final != value:
```

Comment:
The first check is boolean check (is instant) whereare comparison of values is slower, and because if `are_same` is already set to `False` it does not make sense to check values, it is actually faster this way.

Reply:
You are completely right!

If we're optimizing anyway. Note that if:
- final is not None
- not final
- not are same

You can actually opt out early since there's no more need for doing intersections and you already know values are not the same.

Which means you could do:
```python
if final is None:
    final = value
elif not are_same and not final:
    return final, not are_same
elif not are_same or final != value:
    ...
```